app/javascript/components/general/general_post.jsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const GeneralPost = ({ image, header, text, buttonText, link, positionImageRight }) => (
+  <div className="general-component-card fgs-al fgs-al-v fgs-al-g-30">
+    <div className="content-block fgs-al fgs-ali fgs-al-h fgs-al-g-60 fgs-al-align-items-center fgs-al-justify-content-space-evenly">
+      <div className={`image-block fgs-ali ${positionImageRight ? 'right' : ''}`}>
+        <img className="card-image" src={image} alt="general component" />
+      </div>
+
+      <div className="content fgs-al fgs-ali fgs-al-v fgs-al-g-30 ">
+        <div className="title fgs-ali heading-regular fgs-al-g-16">{header}</div>
+        <p className="text fgs-ali">{text}</p>
+        <a href={link}>
```

Comment:
@NRaeewal I think we don't need a link to wrap a button. What about passing onClick as a Prop to manage button functionality...? Also, I would pass onHide element to hide the button if we don't need it...

Reply:
@NRaeewal let's add a ButtonClass prop too to change button style if we want too...